client/src/util/typedCrossfilter/sort.js

```diff
@@ -86,15 +168,255 @@ function quicksortIndirect(a, s, lo, hi) {
   return a;
 }
 
-// Convenience wrappers
+function quicksortFloatsIndirect(a, s, lo, hi) {
+  if (hi - lo < SmallArray) {
+    return insertionsortFloatsIndirect(a, s, lo, hi);
+  }
+  if (lo < hi) {
+    // partition
+    const mid = Math.floor((lo + hi) / 2);
+    const p = a[mid];
+    const t = s[p];
+    let i = lo - 1;
+    let j = hi + 1;
+    while (i < j) {
+      do {
+        i += 1;
+      } while (lt(s[a[i]], t));
+      do {
+        j -= 1;
+      } while (gt(s[a[j]], t));
+      if (i < j) {
+        const tmp = a[i];
+        a[i] = a[j];
+        a[j] = tmp;
+      }
+    }
+    // sort
+    quicksortFloatsIndirect(a, s, lo, j);
+    quicksortFloatsIndirect(a, s, j + 1, hi);
+  }
+  return a;
+}
+
+/*
+Convenience wrappers, handling optimization paths and default
+handlers for NaN comparisons.
+*/
 export function sort(arr, comparator = undefined) {
   if (comparator !== undefined) {
-    // XXX for now
     return arr.sort(arr, comparator);
```

Comment:
is this the same issue with the standard sort method (in place and only comparator)?

Reply:
yes, but I also removed this code path, as it was unused.